### PR TITLE
simplify implementation of `Session.unsubscribe()`

### PR DESCRIPTION
**Pull request by ericvw**
This makes the implementation of details of `Session.unsubscribe()` more
functional in nature and constrains the roles of responsibility of the
sequence of events that occur.

### CODE REVIEW

**Comment by yupinghu** [comment]
LGTM.

**Comment by ericvw** [comment]
Updated commit to address #115 as well it turns out.  Commit message has been updated appropriately.

File: lib/blpapi-wrapper.ts [hunk 1]
```diff
@@ -467,23 +466,33 @@ export class Session extends events.EventEmitter {
...
+        });
+
+        this.session.unsubscribe(subs);
+
+        _.forEach(cids, (cid: number): void => {
+            process.nextTick((): void => {
+                this.subscriptions[cid].emit('end');
+                delete this.subscriptions[cid];
+            });
+            this.unlisten('MarketDataEvents', cid);
```
Comment (thmiceli) [inline lib/blpapi-wrapper.ts:495]: Why 'MarketDataEvents'? Shouldn't this be depending on the subscribed service?

Comment (ericvw) [inline lib/blpapi-wrapper.ts:495]: I kept the existing behavior and left the fix already in #134 to address the issue you mentioned.

**Comment by yupinghu** [comment]
LGTM

File: lib/blpapi-wrapper.ts (hunk 1 above)
Comment (thmiceli) [inline lib/blpapi-wrapper.ts:495]: Ok, LGTM then.

**Comment by ericlu88** [comment]
LGTM.

**Pull request merged**